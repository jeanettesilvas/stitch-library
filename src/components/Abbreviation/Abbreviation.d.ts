/// <reference types="react" />
import './Abbreviation.scss';
interface AbbreviationProps {
    className?: string;
    text: string;
}
export declare const Abbreviation: (props: AbbreviationProps) => JSX.Element;
export {};
