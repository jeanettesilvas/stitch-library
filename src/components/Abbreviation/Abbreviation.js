import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import './Abbreviation.scss'

export class Abbreviation extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    text: PropTypes.string.isRequired,
  }

  render() {
    return (
      <div className={`abbreviation ${this.props.className}`}>
        {this.props.text}
      </div>
    )
  }
}
