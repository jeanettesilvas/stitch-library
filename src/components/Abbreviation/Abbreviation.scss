@import '../../style/global.scss';

.abbreviation {
  background-color: #fff;
  border: 4px solid $gray-e;
  color: #333;
  display: inline-block;
  font-family: $base-font;
  font-size: 22px;
  font-weight: 800;
  padding: 4px $spacing-2x;
}
