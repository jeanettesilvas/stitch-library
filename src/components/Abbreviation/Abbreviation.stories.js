import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs, text } from '@storybook/addon-knobs'

import { Abbreviation } from './Abbreviation'

storiesOf('Abbreviation', module)
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div style={{ marginTop: '50px', textAlign: 'center' }}>{story()}</div>
  ))
  .add('abbr', () => (
    <Abbreviation text={text('abbreviation', 'abbr', 'abbr')} />
  ))
