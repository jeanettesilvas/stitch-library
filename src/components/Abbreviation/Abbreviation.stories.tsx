import React from 'react'
import { Story, Meta } from '@storybook/react'

import { Abbreviation, AbbreviationProps } from './Abbreviation'

export default {
  title: 'Components/Abbreviation',
  component: Abbreviation,
} as Meta

const Template: Story<AbbreviationProps> = (args) => <Abbreviation {...args} />

export const SingleCrochet = Template.bind({})
SingleCrochet.args = {
  text: 'sc',
}

export const DoubleCrochet = Template.bind({})
DoubleCrochet.args = {
  text: 'dc',
}

export const SlipStitch = Template.bind({})
SlipStitch.args = {
  text: 'slst',
}
