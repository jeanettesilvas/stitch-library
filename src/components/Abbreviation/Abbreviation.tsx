import './Abbreviation.scss'

export interface AbbreviationProps {
  className?: string
  text: string
}

export const Abbreviation = (props: AbbreviationProps) => {
  let { className, text } = props

  return <div className={`abbreviation ${className}`}>{text}</div>
}
