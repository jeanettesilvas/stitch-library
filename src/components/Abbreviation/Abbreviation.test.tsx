import renderer from 'react-test-renderer'

import { Abbreviation } from './Abbreviation'

describe('Abbreviation', () => {
  it('renders', () => {
    const tree = renderer.create(<Abbreviation text="sc" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
