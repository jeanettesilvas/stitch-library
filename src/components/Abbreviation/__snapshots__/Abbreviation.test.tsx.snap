// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abbreviation renders 1`] = `
<div
  className="abbreviation undefined"
>
  sc
</div>
`;
